crd/clustersubnetstate/api/v1alpha1/clustersubnetstate.go

```diff
@@ -0,0 +1,49 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// ClusterSubnetState is the Schema for the ClusterSubnetState API
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Timestamp",type=string,JSONPath=`.status.status`
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.spec.timestamp`
+type ClusterSubnetState struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterSubnetStateSpec   `json:"spec,omitempty"`
+	Status ClusterSubnetStateStatus `json:"status,omitempty"`
+}
+
+// ClusterSubnetStateSpec defines the desired state of ClusterSubnetState
+type ClusterSubnetStateSpec struct {
+	Timestamp string `json:"Timestamp"`
+}
+
+// ClusterSubnetStateStatus defines the observed state of ClusterSubnetState
+type ClusterSubnetStateStatus struct {
+	Status bool `json:"Status"`
```

Comment:
if it is just a `bool`, I think we should name this field something descriptive such as `Exhausted`, but I am open to suggestions